### DISCUSSION TITLE
getStaticPaths and fallback: true

### DISCUSSION THREAD

**Discussion by popeating** [vercel/next.js#34317] [Help]
Following the documentation, i enabled fallback from page generation with getStaticPath
the page i'm using is [category]/[slug].js and im also using a multilanguage configuration:
```
module.exports = {
..
  i18n: {
    locales: ['it', 'en'],
    defaultLocale: 'it',
  },
};
```

my [category]/[slug].js page looks something like:
```
const Slug = ({ postData, menus, preview }) => {
  const router = useRouter;
  if (router.isFallback && !postData) {
    return <p>loading...</p>;
  }
  return (
  <>RETURN CODE</>
  )
};
export default Slug;

export async function getStaticPaths({ locales }) {
  const allPosts = await getAllPostsWithSlug();
  const paths = allPosts.edges
    .map(({ node }) =>
      locales.map((locale) => ({
        params: {
          category: node.categories.edges[0].node.slug,
          slug: node.slug,
        },
        locale,
      }))
    )
    .flat();
  return {
    paths,
    fallback: true,
  };
}

export async function getStaticProps({
  params,
  preview = false,
  previewData,
  locale,
}) {
  const data = await getPost(params.slug, preview, previewData, locale);
  const menus = await getPrimaryMenu();
  return {
    props: {
      preview,
      postData: data.post,
      menus,
    },
    revalidate: 10,
  };
}
```

when i try to build `npm run build` at the point the building is trying to create pages it gave me this error (_feturedImage_ should be present in postData):
```
[    ] info  - Generating static pages (0/28)
Error occurred prerendering page "/it/[category]/[slug]". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'featuredImage')
```

it looks like the build process try execute the rendering process at the beginning before using getStaticPaths and getStaticProps (if i print some logs inside getStaticProps, they apper after the error) or maybe getStaticProps before getStaticPatsh
i also tested **getAllPostsWithSlug** to be sure, and it return the correct values, no empty results at the beginning, also paths variable is correct

if use `fallback: 'blocking'` the build success, but i prefere the true option
what am i doing wrong?

**Comment by wwwdepot** [comment]
Error is pretty much self-explanatory. React component needs to handle when `featuredImage = undefined`.

```jsx
{ postData.featuredImage  ? <RenderFeatureImage ... /> : null }
```

If your own data does not have this column, remove the matching component.

**Comment by popeating** [comment]
i see what's happening, but featuredImage should never be empty, unless the page is rendered before the getStaticProps (which im really sure return feturedImage)

anyway, for the moment i reduced the page, but it looks the `fallback: true` is not working correctly with multilange site on default language; the build process build the pages with language prefix path, even for default, so when i request a page that is not yet generated for default language (lets say /page/post-01) it is not found in the list of the fallback (/en/page/post-01) so it is generated, but no router.isFallback is set
